CONTRIBUTING.md

```diff
@@ -1,84 +1,80 @@
 
-This doc is intended for contributors to Fx (hopefully that's you!).
+Thanks for helping to make Fx better for everyone!
 
-## Development environment
+If you'd like to add new exported APIs, please [open an issue][open-issue]
+describing your proposal &mdash; discussing API changes ahead of time makes
+pull request review much smoother.
 
-* **Go**. Install on OS X with `brew install go`. Make sure `go version` returns at
-  least `1.8` since we're going to be using 1.8+ features like subtests.
+Note that you'll need to sign [Uber's Contributor License Agreement][cla]
+before we can accept any of your contributions. If necessary, a bot will remind
+you to accept the CLA when you open your pull request.
 
-* **[overcommit](https://github.com/brigade/overcommit)**, a git hook manager.
-  Install `overcommit` into your path with `sudo gem install overcommit`.
-  Enable it on the Fx repo with `overcommit --install`.
-  We use overcommit to enforce a variety of style, semantic, and legal things
-  (for example, license headers on all source files).
+## Setup
 
-## Checking out the code
+[Fork][fork], then clone the repository:
 
-Make sure the repository is cloned to the correct location:
-
-```bash
-go get go.uber.org/fx/...
-cd $GOPATH/src/go.uber.org/fx
+```
+mkdir -p $GOPATH/src/go.uber.org
+cd $GOPATH/src/go.uber.org
+git clone git@github.com:your_github_username/fx.git
+cd fx
+git remote add upstream https://github.com/uber-go/fx.git
+git fetch upstream
 ```
 
-## Dependency management
-
-Dependencies are tracked via `glide.yaml`. If you're not familiar with `glide`,
-[read the docs](https://github.com/Masterminds/glide#usage).
-
-## License headers
-
-This project is open source software and requires a header at the beginning of
-every source file. This is enforced by commit hooks and TravisCI.
-
-To add license headers, use
-[update-licence](go.uber.org/tools/update-license), which is included in
-`make dependencies`.
-
-## Commit Messages
-
-Overcommit adds some requirements to your commit messages. At Uber, we follow the
-[Chris Beams](http://chris.beams.io/posts/git-commit/) guide to writing Git
-commit messages. Read it, follow it, learn it, love it.
+Install Fx's dependencies:
 
-## FIXMEs
+```
+make dependencies
+```
 
-If you ever are in the middle of writing code and remember a thing you need to
-do, leave a comment like:
+Make sure that the tests and the style checkers pass:
 
-```go
-// FIXME(ai) make this actually work
+```
+make test
+make lint
 ```
 
-Your initials in parens are optional but good practice. This is better
-than a TODO because our CI checks for unresolved FIXMEs. If you forget to fix
-a FIXME, your code won't get merged.
+If you're not using the minor version of Go specified in the Makefile's
+`LINTABLE_MINOR_VERSIONS` variable, `make lint` doesn't do anything. This is
+fine, but it means that you'll only discover style violations after you open
+your pull request.
 
-## Testing
+## Making Changes
```

Comment:
Ya, GitHub lacks a real set standard on style like this, but it definitely leans toward H1s = title, H2s and below = sentence. I'm probably the only one who cares, but I'm for embracing this rule consistently.